cmd/cluster/core/dump.go

```diff
@@ -201,43 +252,10 @@ func dumpGuestCluster(ctx context.Context, opts *DumpOptions) error {
 		}
 	}()
 
-	target := opts.ArtifactDir + "/hostedcluster-" + opts.Name
-
-	podToForward, err := supportforwarder.GetRunningKubeAPIServerPod(ctx, c, cpNamespace)
-	if err != nil {
-		return fmt.Errorf("failed to get running kube-apiserver pod for guest cluster: %w", err)
-	}
-
-	restConfig, err := util.GetConfig()
-	if err != nil {
-		return fmt.Errorf("failed to get a config for management cluster: %w", err)
-	}
-
-	if len(opts.ImpersonateAs) > 0 {
-		restConfig.Impersonate = restclient.ImpersonationConfig{
-			UserName: opts.ImpersonateAs,
-		}
-	}
-
-	kubeClient, err := kubeclient.NewForConfig(restConfig)
-	if err != nil {
-		return fmt.Errorf("failed to get a kubernetes client: %w", err)
-	}
-	forwarderOutput := &bytes.Buffer{}
-	forwarder := supportforwarder.PortForwarder{
-		Namespace: podToForward.Namespace,
-		PodName:   podToForward.Name,
-		Config:    restConfig,
-		Client:    kubeClient,
-		Out:       forwarderOutput,
-		ErrOut:    forwarderOutput,
-	}
-	podPort := supportutil.KASPodPortFromHostedCluster(hostedCluster)
-	forwarderStop := make(chan struct{})
-	if err := forwarder.ForwardPorts([]string{fmt.Sprintf("%d:%d", localPort, podPort)}, forwarderStop); err != nil {
-		return fmt.Errorf("cannot forward kube apiserver port: %w, output: %s", err, forwarderOutput.String())
+	target := filepath.Join(opts.ArtifactDir, "hostedcluster-"+opts.Name)
+	if err := os.MkdirAll(target, 0755); err != nil {
+		return err
```

Comment:
Can you wrap the err message here? Maybe something like `return fmt.Errorf("failed to create guest cluster dump directory %s: %w", target, err)`

Reply:
Sorry but I don't think that wrapping this error is adding much value.
`err` will already tell that the directory cannot be created and it will tell that with more details.
Also remark that there is no error wrapping for the call to `createGuestKubeconfig`; not saying that this is an excuse to do the same here, but error wrapping should only be done:
- To add more details to an existing error
- Or when we want to change the type of an error